File: design-patterns-2/csharp/patterns/factories/innerFactory/Runner.cs
namespace csharp.patterns.factories.innerFactory;

class Runner : IRunner
{
    public void Run(int exampleNumber)
    {
        var point1 = Point.Factory.CreateCartesianPoint(2, 4);
        var point2 = Point.Factory.CreatePolarPoint(2, 4);

        Console.WriteLine(point1);
        Console.WriteLine("--------------");
        Console.WriteLine(point2);
    }
}

File: design-patterns-2/java/app/src/main/java/org/patterns/patterns/proxy/Runner.java
package org.patterns.patterns.proxy;

import org.patterns.patterns.IRunner;

public class Runner implements IRunner {
    @Override
    public void run(int exampleNum) {
        //Setting some properties 
        Point location = new Point(2, 3);
        ImageFactory factory = new ImageFactory();
        IImage image = factory.getImage("my-img.png");
        image.setLocation(location);

        //Only here we render the image
        image.render();
    }
}
